pkg/route/secretmanager/manager.go

```diff
@@ -29,7 +29,7 @@ type manager struct {
 	// Map of registered handlers for each route.
 	// Populated inside RegisterRoute() and used in UnregisterRoute(), GetSecret.
 	// generateKey() will create the map key.
-	registeredHandlers map[string]secret.SecretEventHandlerRegistration
+	registeredHandlers map[string]*referencedSecret
```

Comment:
Is there a reason to use a pointer type `map[string]*referencedSecret` instead of `map[string]referencedSecret`?

Reply:
Thanks for pointing that out! I've updated the `registeredHandlers` map to use `map[string]referencedSecret` instead of `map[string]*referencedSecret`. Since referencedSecret doesn't need to be modified once it's stored in the map, storing it by value is better.